### ISSUE TITLE
rexml changed status in 3.0

### ISSUE THREAD

**Issue by jrochkind** [ruby-marc/ruby-marc#67]
Ruby 3.0.0 changed how rexml is packaged, the [release notes](https://www.ruby-lang.org/en/news/2020/12/08/ruby-3-0-0-preview2-released/) say: "The following default gems are now bundled gems: rexml". 

I am not totally sure what it means. But i think it may be somehow resulting in any project including ruby-marc in ruby 3.0.0-preview2 erroring on boot with:

```
File does not exist: rexml/parsers/streamparser

rake aborted!
Command failed with status (1)
/opt/hostedtoolcache/Ruby/3.0.0-preview2/x64/bin/bundle:23:in `load'
/opt/hostedtoolcache/Ruby/3.0.0-preview2/x64/bin/bundle:23:in `<main>'
```

Not totally sure what's going on or why, but I think it is ruby-marc. 

My guess is that  `rexml` needs to be explicitly declared as a dependency in gemspec, *or* ruby-marc code needs to be careful about not requiring/using rexml unless it is present/asked for, instead of on boot. 

https://nts.strzibny.name/ruby-stdlib-default-bundled-gems/

https://stdgems.org/

**Comment by billdueber** [comment]
Exactly what is says on the tin: some things that used to come as a part of the standard library have been (pro/de)moted to standalone gems. rubygems.org reports it as already being available as a gem, so the easy thing to do is just declare it as a dependency.

**Comment by jrochkind** [comment]
@billdueber you are back! :)  Do you feel like reviewing/merging https://github.com/ruby-marc/ruby-marc/pull/66 while you are here?

You may not actually be using ruby-marc anymore; I know I am not! Are there any current committers on ruby-marc who currently use it?

**Comment by jrochkind** [comment]
And if I understand it right, rexml will actually still be automatically installed with every ruby installation, it just needs to be declared, say in the gemspec, anyway. 

But the particular error message is VERY WEIRD, not normally the error message you'd get from trying to use a gem, if you weren't trying to require that file specificcally (which this code is not), so that's confusing, but perhaps just a weird artifact of the the way these gems are being slowly decoupled.

**Comment by jrochkind** [comment]
Hm, the downside may be that declaring it as a dependency may actually cause an error in VERY OLD versions of ruby that we still support? We'd want to check that. People may likely still be using ruby-marc with very very old versions of ruby.

**Comment by jrochkind** [comment]
OK, verified that running ruby-marc tests on 3.0.0-preview2 fails with a more expected error `File does not exist: rexml/document`. 

Adding `rexml` to the gemspec takes care of THAT error, but there are still two additional test error/failures in ruby 3.0.0-preview2. 

I'm not sure I have time to debug them. I don't actually use ruby-marc myself at present, I'm just looking at it cause it's a dependency of traject, and I wanted to get traject passing tests on ruby 3.0.0. 

Do we have any active ruby-marc maintainers? Do we need to recruit some? Maybe an email to code4lib list?  I think @cbeer has also volunteered to be added as a committer (although I don't know if that means he's interested in doing work for ruby 3), but I was uncomfortable adding him myself unilaterally -- is there any way to convene a gathering of 2-3 committers to patch the torch? I have been unable to get a single other committer's attention before now, and am not sure if @billdueber is about to disappear again!  Thoughts, anyone?

**Comment by jrochkind** [comment]
(The non-rspec test suite is kind of hard for me to work with, figure out how to run a single test etc, which adds to the burden of figuring out why some tests are failing in ruby 3.0 still)

**Comment by jrochkind** [comment]
(Also wonder if we should just REMOVE rexml support and release a ruby-marc 2.0 with such. It is kind of ancient technology. But that is not necessarily what the current two errors I am seeing are related to).

**Comment by etagwerker** [comment]
@jrochkind Thanks for documenting your issue here! I just ran into this issue when trying to add support to Ruby 3.0 in RubyCritic. 

Adding `rexml` as a development dependency in the gemspec fixed it for me. 😄

**Comment by cbeer** [comment]
Fixed by #69
